scripts/avatars/trops_avatars_seeding.py

```diff
@@ -0,0 +1,164 @@
+
+    # aura avatar deposits
+    trops.balancer.deposit_and_stake(
+        [wbtc, badger],
+        bpt_ratios_amounts(
+            bpt_badgerwbtc, wbtc_rate, badger_rate, wbtc_decimals, badger_decimals
+        ),
+        pool=bpt_badgerwbtc,
+        stake=False,
+    )
+
+    weth_mantissa = int(((FUNDS_PER_POSITION // 2) / weth_rate) * 1e18 / 0.98)
+    trops.balancer.swap(weth, reth, weth_mantissa, pool=bpt_wethreth)
+    trops.balancer.deposit_and_stake(
+        [badger, reth],
+        bpt_ratios_amounts(
+            bpt_badgerreth, badger_rate, reth_rate, badger_decimals, reth_decimals
+        ),
+        pool=bpt_badgerreth,
+        stake=False,
+    )
+
+    # NOTE: for `bpt_wbtcdigggravi` acquisition prefer to avoid 2 swaps and simply deposit full amount in wbtc directly
+    wbtc_amount = (FUNDS_PER_POSITION / wbtc_rate) * 10 ** wbtc_decimals
+    trops.balancer.deposit_and_stake(
+        [wbtc, digg, gravi],
+        [wbtc_amount, 0, 0],
+        pool=bpt_wbtcdigggravi,
+        stake=False,
+    )
+
+    bpt_badgerwbtc_balance = bpt_badgerwbtc.balanceOf(trops)
+    bpt_badgerreth_balance = bpt_badgerreth.balanceOf(trops)
+    bpt_wbtcdigggravi_balancer = bpt_wbtcdigggravi.balanceOf(trops)
+
+    bpt_badgerwbtc.approve(aura_avatar, bpt_badgerwbtc_balance)
+    bpt_badgerreth.approve(aura_avatar, bpt_badgerreth_balance)
+    bpt_wbtcdigggravi.approve(aura_avatar, bpt_wbtcdigggravi_balancer)
+    aura_avatar.deposit(
+        aura_avatar.getPids(),
+        [bpt_badgerwbtc_balance, bpt_wbtcdigggravi_balancer, bpt_badgerreth_balance],
+    )
+
+    # convex avatar deposit
+    trops.curve.deposit_zapper(
+        fraxbp_zap,
+        pool_fraxbp,
+        [badger, frax, usdc],
+        [
+            ((FUNDS_PER_POSITION // 2) / badger_rate) * 1e18,
+            0,
+            (FUNDS_PER_POSITION // 2) * 1e6,
+        ],
+    )
+
+    lp_balance = badger_fraxbp.balanceOf(trops)
+    badger_fraxbp.approve(wcvx_badger_fraxbp, lp_balance)
+    wcvx_badger_fraxbp.deposit(lp_balance, trops)
+
+    staking_lp_balance = wcvx_badger_fraxbp.balanceOf(trops)
+    assert staking_lp_balance == lp_balance
+
+    wcvx_badger_fraxbp.approve(convex_avatar, staking_lp_balance)
+    convex_avatar.depositInPrivateVault(
+        convex_avatar.getPrivateVaultPids()[0], staking_lp_balance, False
+    )
+
+    trops.post_safe_tx()
+
+
+def add_members():
+    techops = GreatApeSafe(r.badger_wallets.techops_multisig)
+
+    upkeep_manager = techops.contract(r.badger_wallets.upkeep_manager)
+
+    upkeep_manager.addMember(
+        r.avatars.aura,
+        "AuraAvatar",
+        # gas figure ref: https://github.com/Badger-Finance/badger-multisig/issues/1207#issuecomment-1490226452
+        990_000,
+        0,
+    )
+
+    upkeep_manager.addMember(
+        r.avatars.convex,
+        "ConvexAvatar",
+        # gas figure ref: https://github.com/Badger-Finance/badger-multisig/issues/1207#issuecomment-1490226452
+        5_007_000,
```

Comment:
oof 5m, i hope that 4eth max from gas station will cover this.

not sure how you got to 5m though? your gas snapshot shows 2.5m for two pids? you doubled both in order to support up to 6 instead of 3 on aura and 4 instead of 2 on convex?

Reply:
getting `VirtualMachineError: revert: typed error: 0x649bf810` again on this line, even after topping up with 100eth from `a[0]`.

is 5m too much for the upkeep mngr, no matter how much eth it is holding?

Reply:
yes, doubled the current quantity that they are supposed to hold and test were run with (3 for aura and 2 for convex):

- aura: 165000 * 6 (will support total of 6 pids) = 990000
- convex: 1251727 * 4 ("" of 4 pids) = 5006908

Reply:
have not realised of this [limitation > 5M](https://etherscan.io/address/0x02777053d6764996e594c3E88AF1D58D5363a2e6#code#F1#L732), which causes the revert. in theory we will need ~1.86eth for that gas limit, but we are restricted with that condition. i decreased to 5m in cb82c6ee36b263e020659d26b5ea74db1883a9c5

Reply:
maybe add a ticket to the upkeep manager repo to add a check there, based on the `s_storage.maxPerformGas`?

Reply:
added as a comment in the existing [ticket](https://github.com/Badger-Finance/badger-avatars/issues/30)